comments/main.py

```diff
@@ -0,0 +1,234 @@
+    visual = checkers_board.generate_visual_with_flashing(flashing_position)
+    piece_number = current_piece.name.split("_")[-1]
+    print(f"      {current_piece.team.capitalize()} Players Turn")
+    print("   " + "-" * 24)
+    print(visual)
+    print(f"{current_piece.team.capitalize()} Piece {piece_number} at {current_piece.xy_coord}")
+
+"""
+Piece Logic
+"""
+
+def show_valid_moves(valid_moves): # Fancy Print Statement for valid_moves
+    show_choices = []
+    for name, value in valid_moves.items():
+        if value:
+            show_choices.append(name)
+    return show_choices
+
+
+def process_move(valid_moves, piece, capture=False):
+    player_options = show_valid_moves(valid_moves)
+    print(f"Here are your valid moves: {player_options}")
+    user_input = input("Make a choice: ")
+    
+    if (user_input in valid_moves) and (capture is True): # If you can capture do so
+        selected_move = valid_moves[user_input]
+        capture_coord = piece.moves[user_input]
+        # print(f"Debug capture {capture_coord}")
+        checkers_board.remove_piece_from_game(capture_coord)  # Remove captured Piece from
+        checkers_board.move(selected_move, piece)  # Move Player piece to new coord
+        checkers_board.remove_from_location(piece.xy_coord) # Remove Player piece from old coord
+        # print(f"Debug selected_move {selected_move}")
+        # print(f"Debug piece.moves {piece.moves}")
+        piece.xy_coord = selected_move  # Update (x, y) of piece.xy_coord attr
+        # print(f"debug of piece.xy_coord {piece.xy_coord}")
+        piece.check_valid_move()  # Update piece.moves to reflect new possible moves
+        checkers_board.check_king_me(piece)  # Update piece.is_king if possible
+        is_game_over() # Check for Victory Condition
+        # print(f"Debug of updated piece.moves {piece.moves}") 
+
+        """
+        currently a bug where capture flag is incorrectly made false when it should be 
+        true. this causes the below elif statement to trigger and the piece during what 
+        should be a capture move instead normally moves into the space, but it is occupied by the opp_piece
+        """
+
+    elif user_input in valid_moves and (capture is False):  # If you can't capture, move regularly
+        print(f"You choice was to move to {user_input}")
+        selected_move = piece.moves[user_input]
+        checkers_board.move(selected_move, piece)  # Move piece to (x, y) location
+        checkers_board.remove_from_location(piece.xy_coord)  # Remove piece from board at its previous (x, y)
+        piece.xy_coord = selected_move  # Update (x, y) of piece.xy_coord attr
+        piece.check_valid_move()  # Update piece.moves to reflect new possible moves
+        checkers_board.check_king_me(piece)  # Update piece.is_king if possible
+    else:
+        print("Invalid Move. Try Again")
+        process_move(valid_moves, piece)
+
+
+def check_capturable_moves(selected_piece, team):
```

Comment:
In general there is a lot going on here. I don't think what each checker_board function returns is very straight forward.

After going through your board code and what it returns I get it... But you're using dictionaries in a non-conventional way that gets a little confusing.

can_this_move is a dictionary of potential moves for either white or black piece. And the values are if this piece can move in that direction which is set to True or False.

To determine if it can move in that direction or not in is_regular_move_valid() you check what piece color it is. then you check if the piece can move in that direction by seeing if the piece.moves[...] direction is NOT None (if its None you've hit the end of the board). Then you check if there is a another game piece in this location by calling get_from_location(). Then if there isn't you set the move to True. Otherwise the moves you're returning in is_regular_move_valid() is False.

After you get this dictionary can_this_move you iterate through the moves and you check again if there is a piece there by checking to see if the move is False or True. If its False you get the coordinates. If you fail to get the coordinates you assume the piece.moves[] there is a wall and you're done. If NOT then you check what is in that coordinate space. If the piece there is the opposite team then you check if you can capture this piece by checking the piece behind it.

The problem is this... You already checked the move inside of is_regular_move_valid. Then you're checking it again out here. You're doing things twice again. You should be able to pass some more data back so you don't have to do the logic twice.